fs: TestAPIAbortMultipartHandler sporadically fails under Windows

Below is the log of the failed test: (I added a log to print the real cause of the error)

```
Unable to abort multipart upload. remove \\?\c:\Users\appveyor\AppData\Local\Temp\1\minio-412581580\.minio.sys\multipart\1ks3s3fokor62m02gaeqhc9apmkxwes6qpv933o5fx4pbk9dex5p9mpszry5\test-object-new-multipart\6902e697-4b13-4771-9d3f-489a1703053c\object2: The process cannot access the file because it is being used by another process.
--- FAIL: TestAPIAbortMultipartHandler (0.45s)
	test-utils_test.go:1759: Minio FS: Anonymous HTTP request test fail for "TestAPIAbortMultipartHandler": 
		<Error> Expected the anonymous HTTP request to be served after the policy changes
		,Expected response HTTP status code to be 204, got 500
```

In Windows, I can reproduce this error ```: The process cannot access the file because it is being used by another process.``` with the following code:

```
package main

import "log"
import "os"

func main() {	
	file, err := os.Open("myfile") // For read access.
	if err != nil {
		log.Fatal(err)
	}
	if err := os.Remove("myfile"); err != nil {
		log.Fatal(err)
	}
	file.Close()
}

c:\test>go run test.go
2016/11/25 23:14:08 remove myfile: The process cannot access the file because it is being used by another process.
exit status 1
```

The sample code opens a file and tries to remove it before closing the fd.

So, it looks like the problem is that we are aborting the multipart upload (which means deleting multipart files, https://github.com/minio/minio/blob/master/cmd/fs-v1-multipart.go#L709) while appendPart is still not finished (appendPart is reading from the part and appends to a tmp file, https://github.com/minio/minio/blob/master/cmd/fs-v1-background-append.go#L196 )

What do you think ?

--- a/cmd/fs-v1-background-append.go
+++ b/cmd/fs-v1-background-append.go
@@ -115,13 +115,14 @@ func (b *backgroundAppend) complete(disk StorageAPI, bucket, object, uploadID st
 // Called after complete-multipart-upload or abort-multipart-upload so that the appendParts go-routine is not left dangling.
 func (b *backgroundAppend) abort(uploadID string) {
 	b.Lock()
-	defer b.Unlock()
 	info, ok := b.infoMap[uploadID]
 	if !ok {
+		b.Unlock()
 		return
 	}
 	delete(b.infoMap, uploadID)
-	close(info.abortCh)
+	b.Unlock()
+	info.abortCh <- struct{}{}
 }
 
 // This is run as a go-routine that appends the parts in the background.

--- a/cmd/fs-v1-multipart.go
+++ b/cmd/fs-v1-multipart.go
@@ -637,11 +637,12 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 // the directory at '.minio.sys/multipart/bucket/object/uploadID' holding
 // all the upload parts.
 func (fs fsObjects) abortMultipartUpload(bucket, object, uploadID string) error {
+	// Signal appendParts routine to stop waiting for new parts to arrive.
+	fs.bgAppend.abort(uploadID)
 	// Cleanup all uploaded parts.
 	if err := cleanupUploadedParts(bucket, object, uploadID, fs.storage); err != nil {
 		return err
 	}
-	fs.bgAppend.abort(uploadID)
 	// remove entry from uploads.json with quorum
 	if err := fs.removeUploadID(bucket, object, uploadID); err != nil {
 		return toObjectErr(err, bucket, object)

